cpp/src/io/orc/stripe_enc.cu

```diff
@@ -764,10 +764,7 @@ extern "C" __global__ void __launch_bounds__(512)
             int32_t ts_scale = kTimeScale[min(s->chunk.scale, 9)];
             int64_t seconds  = ts / ts_scale;
             int32_t nanos    = (ts - seconds * ts_scale);
-            if (nanos < 0) {
-              seconds += 1;
-              nanos += ts_scale;
-            }
+            if (nanos < 0) { nanos += ts_scale; }
```

Comment:
I don't get this change, seems like the right thing to do would be to reduce seconds by one. Can you please add a comment?

Reply:
This was weird to me too because -2.6 seconds would be encoded as -2 s and 4x10^8 ns. This doesn't seem to make sense because if nanoseconds have to be positive then the encoding should be -3 s , 4x10^8 ns so that the second component + ns component gives us the actual timestamp. But apache's **reader** implementation seems to be working with the understanding that if seconds component is negative then reduce it by one.

https://github.com/apache/orc/blob/fa9c011e13e8376d2a185bd76af834bd644f4332/java/core/src/java/org/apache/orc/impl/TreeReaderFactory.java#L1221-L1227

Reply:
How about I link this comment there? https://github.com/rapidsai/cudf/pull/5529#issuecomment-648768925

Reply:
Sounds good :+1: !
This turned out to be much more complex than it looks at first glance :D